src/python/pants/backend/javascript/package_json.py

```diff
@@ -490,7 +495,28 @@ async def read_package_jsons(globs: PathGlobs) -> PackageJsonForGlobs:
 
 @rule
 async def all_package_json() -> AllPackageJson:
```

Comment:
It would be good to add a note here that explains why this avoids using the more obvious APIs.